build/source/engine/read_attrb.f90

```diff
@@ -29,304 +29,328 @@ module read_attrb_module
  ! ************************************************************************************************
  ! public subroutine read_dimension: read HRU and GRU dimension information on local attributes
  ! ************************************************************************************************
- subroutine read_dimension(maxGRU,maxHRU,err,message)
+ subroutine read_dimension(attrFile,fileGRU,fileHRU,nGRU,nHRU,err,message,startGRU,checkHRU)
```

Comment:
I think we had gone through discussion that we wanted to separate allocation and index assignment.